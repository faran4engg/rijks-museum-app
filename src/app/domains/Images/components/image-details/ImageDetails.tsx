import { FC } from 'react';
import { ImageDetailsProps } from './types';

const ImageDetails: FC<ImageDetailsProps> = ({ authorName, title }) => (
  <div
    id="img-details"
    className="w-48 mx-auto mb-4 -mt-6 overflow-hidden rounded-lg shadow-lg lg:-mt-8 lg:w-56"
  >
    <h3 className="py-1 text-xs font-bold tracking-wide text-center text-gray-800 capitalize bg-white lg:py-2 dark:bg-gray-800 dark:text-white ">
      <span>ðŸŽ¨&nbsp;</span>
      {authorName}
    </h3>

    <div className="flex items-center justify-center px-2 py-1 bg-gray-200 lg:py-2 dark:bg-gray-700">
      <span className="text-xs font-bold text-center text-gray-800 dark:text-gray-200 line-clamp-1 hover:line-clamp-none">
        <span>ðŸ“Œ&nbsp;</span>
        {title}
      </span>
    </div>
  </div>
);

export default ImageDetails;
