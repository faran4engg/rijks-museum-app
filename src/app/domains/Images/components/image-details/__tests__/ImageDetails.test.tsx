import { create } from 'react-test-renderer';

import ImageDetails from '../ImageDetails';

describe('ImageDetails', () => {
  it('should render', () => {
    const component = create(
      <ImageDetails authorName="faran" title="my painting" />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
