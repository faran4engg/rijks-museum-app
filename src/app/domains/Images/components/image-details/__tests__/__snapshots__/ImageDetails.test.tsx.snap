// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDetails should render 1`] = `
<div
  className="w-48 mx-auto mb-4 -mt-6 lg:-mt-10 overflow-hidden rounded-lg shadow-lg lg:w-56"
  id="img-details"
>
  <h3
    className="py-1 lg:py-2 text-xs capitalize font-bold tracking-wide text-center text-gray-800 bg-white dark:bg-gray-800 dark:text-white "
  >
    <span>
      ðŸŽ¨Â 
    </span>
    faran
  </h3>
  <div
    className="flex items-center justify-center px-2 py-1 lg:py-2 bg-gray-200 dark:bg-gray-700"
  >
    <span
      className="text-xs font-bold text-center text-gray-800 dark:text-gray-200 line-clamp-1 hover:line-clamp-none"
    >
      <span>
        ðŸ“ŒÂ 
      </span>
      my painting
    </span>
  </div>
</div>
`;
